src/nlp/utils/file_utils.py

```diff
@@ -213,10 +212,9 @@ def cached_path(url_or_filename, download_config=None, **download_kwargs,) -> Op
             return output_path
 
         # Path where we extract compressed archives
-        # We avoid '.' in dir name and add "-extracted" at the end: "./model.zip" => "./model-zip-extracted/"
-        output_dir, output_file = os.path.split(output_path)
-        output_extract_dir_name = output_file.replace(".", "-") + "-extracted"
-        output_path_extracted = os.path.join(output_dir, output_extract_dir_name)
+        # We extract in the cache dir, and get the extracted path name by hashing the original path"
+        abs_output_path = os.path.abspath(output_path)
+        output_path_extracted = os.path.join(cache_dir, hash_url_to_filename(abs_output_path))
```

Comment:
This should fix the collision problem when extracting in the cache dir @patrickvonplaten

Reply:
looks great!